lib/services/screen-builder-service.ts

```diff
@@ -77,34 +119,71 @@ export class ScreenBuilderService implements IScreenBuilderService {
 		}).future<IScreenBuilderOptions>()();
 	}
 
-	private prepareScreenBuilder(): IFuture<void> {
+	public promptGenerate(generatorName: string, screenBuilderOptions?: IScreenBuilderOptions): IFuture<IScaffolder> {
 		return (() => {
-			this.$appScaffoldingExtensionsService.prepareAppScaffolding().wait();
+			let scaffolderData = this.createScaffolder(generatorName, screenBuilderOptions).wait();
+			let scaffolder = scaffolderData.scaffolder;
+			let type = screenBuilderOptions.type || ScreenBuilderService.DEFAULT_SCREENBUILDER_TYPE;
+			type = ScreenBuilderService.PREDEFINED_SCREENBUILDER_TYPES[type] || type;
+
+			if (type === ScreenBuilderService.DEFAULT_SCREENBUILDER_TYPE || screenBuilderOptions.answers) {
+					scaffolder.promptGenerate(type, screenBuilderOptions.answers, scaffolderData.callback);
+			} else {
+					scaffolder.promptGenerate(type, scaffolderData.callback);
+			}
 
-			let generators = this.$dependencyConfigService.getAllGenerators().wait();
-			_.each(generators, (generator: IGeneratorConfig) => this.$generatorExtensionsService.prepareGenerator(generator.name).wait());
+			return scaffolderData;
+		}).future<IScaffolder>()();
+	}
+
+	public ensureScreenBuilderProject(projectDir: string): IFuture<void> {
+		return (() => {
+			if(!_.every(ScreenBuilderService.SCREEN_BUILDER_SPECIFIC_FILES, file => this.$fs.exists(path.join(projectDir, file)).wait())) {
+				this.$errors.fail("This command is applicable only to Screen Builder projects.");
```

Comment:
Should this be fail or failWithoutHelp? Which one would be more helpful to the user?